data/import_data.py

```diff
@@ -16,83 +18,139 @@ async def fetch_xml(session, url):
-                "kod": c.xpath("kod/text()")[0],
-                "nazev": c.xpath("nazev/text()")[0],
-                "kredity": int(c.xpath("kredity/text()")[0]),
-                "rozsah": rozsah,
-                "rozvrhy": [],
-                "zpuszak": c.xpath("zpuszak/text()")[0],
-                "anotace": anotace,
-                "osnova": osnova,
-                "pozadavky": pozadavky,
-                "osnova_cv": osnova_cv,
-                "cile": cile,
-                "obor": obor
-            })
-
-    OBORY = list(sorted((o for o in OBORY if o.startswith("BS ")), key=alphas)) + \
-            list(sorted((o for o in OBORY if not o.startswith("BS ")), key=alphas))
-
-    OBORY = [fixup_name(o) for o in OBORY]
-
-    print("Vyskytu predmetu celkem: " + str(len(BK_COURSES1)))
-
-    BK_COURSES_GROUPED = defaultdict(lambda: [])
-    for c in BK_COURSES1:
-        BK_COURSES_GROUPED[c["predmet_id"]].append(c)
-    print("Jedinecnych predmetu: " + str(len(BK_COURSES_GROUPED)))
-
-    BK_COURSES2 = {}
-    for k, v in BK_COURSES_GROUPED.items():
-        c = dict(v[0])
-        del c["obor"]
-
-        BK_COURSES2[k] = c
-
-    f = open("data.js", "w", encoding="utf-8")
-    f.write("export const OBORY=" + tojson(OBORY) + "\n\n")
-    f.write("export const COURSES=" + tojson(BK_COURSES2) + "\n")
-    f.close()
-
-
-async def fetch_study_plans(urls):
+def fixup_specialization_name(s):
+    s = re.sub(" +", " ", s)
+    return s.replace("jaderné", "Jaderné")
+
+
+def subject_codes(document):
+    return document.xpath("/definitions/courses/course/acronym/text()")
+
+
+def get_course(document, course_acronym):
+    course_elements = document.xpath('/definitions/courses/course[acronym/text()="{}"]'.format(course_acronym))
+    return max(course_elements, key=lambda course: len(unique_course_time_slots(course)))
+
+
+def course_lectures(course):
+    return course.xpath('/definitions/lectures/lecture[course/@ref={}]'.format(course.attrib['id']))
+
+
+def lecture_part_ids(lecture):
+    xpath = '/definitions/cards/card/part[parent::card[lecture[@ref="{}"]]]/@ref'.format(lecture.attrib['id'])
+    return {int(x) for x in lecture.xpath(xpath)}
+
+
+def part_ids_for_course(course):
+    return set().union(*(lecture_part_ids(lecture) for lecture in course_lectures(course)))
+
+
+def is_card_valid(card):
+    return card.xpath("string(day/@ref)") != '' and card.xpath("string(time/@ref)") != ''
+
+
+def card_data(card):
+    lecture_id = card.xpath("string(lecture/@ref)")
+    lecturer_id = card.xpath("string(lecturer/@ref)")
+    classroom_id = card.xpath("string(classroom/@ref)")
+    begin = int(card.xpath("string(day/@ref)")) * 100 + int(card.xpath("string(time/@ref)")) + 6.5
+    end = begin + int(card.xpath("string(/definitions/lectures/lecture[@id=\"{}\"]/duration)".format(lecture_id)))
+    what = "C" if card.xpath("string(/definitions/lectures/lecture[@id={}]/tag)".format(lecture_id)) in ["ex",
```

Comment:
Pokus o identifikaci, zda se jedná o přednášku nebo cvičení. Potentiálně TODO/FIXME, protože z dat není stoprocentně zřejmé, že takhle je to vždycky korektní, i když většinou by tohle fungovat mělo (v datech je jednak někde přítomný podelement `<tag>cv</tag>` v elementu `<lecture>`, jednak podelement `<practice>1</practice>` v elementu `<course>`, ale jejich korelace, byť silná, není IIRC stoprocentní).